fastchat/serve/huggingface_api_worker.py

```diff
@@ -0,0 +1,383 @@
+    release_worker_semaphore()
+    return JSONResponse(output)
+
+
+@app.post("/worker_get_embeddings")
+async def api_get_embeddings(request: Request):
+    params = await request.json()
+    await acquire_worker_semaphore()
+    embedding = worker.get_embeddings(params)
+    release_worker_semaphore()
+    return JSONResponse(content=embedding)
+
+
+@app.post("/worker_get_status")
+async def api_get_status(request: Request):
+    return worker.get_status()
+
+
+@app.post("/count_token")
+async def api_count_token(request: Request):
+    params = await request.json()
+    return worker.count_token(params)
+
+
+@app.post("/worker_get_conv_template")
+async def api_get_conv(request: Request):
+    return worker.get_conv_template()
+
+
+@app.post("/model_details")
+async def api_model_details(request: Request):
+    return {"context_length": worker.context_len}
+
+
+def create_model_worker():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--host", type=str, default="localhost")
+    parser.add_argument("--port", type=int, default=21002)
+    parser.add_argument("--worker-address", type=str, default="http://localhost:21002")
+    parser.add_argument(
+        "--controller-address", type=str, default="http://localhost:21001"
+    )
+    parser.add_argument(
+        "--supported-models-file", type=str, default="supported_models.json"
+    )
+    parser.add_argument(
+        "--model",
+        type=str,
+        default="falcon-180b-chat",
+        help="The model name to be called.",
+    )
+    parser.add_argument(
+        "--model-names",
+        type=lambda s: s.split(","),
+        default="falcon-180b-chat",
+        help="Optional display comma separated names",
+    )
+    parser.add_argument(
+        "--conv-template", type=str, default=None, help="Conversation prompt template."
+    )
+    parser.add_argument("--embed-in-truncate", action="store_true")
+    parser.add_argument(
+        "--limit-worker-concurrency",
+        type=int,
+        default=5,
+        help="Limit the model concurrency to prevent OOM.",
+    )
+    parser.add_argument("--stream-interval", type=int, default=2)
+    parser.add_argument("--no-register", action="store_true")
+    parser.add_argument(
+        "--seed",
+        type=int,
+        default=None,
+        help="Overwrite the random seed for each generation.",
+    )
+    args = parser.parse_args()
+    logger.info(f"args: {args}")
+
+    with open(args.supported_models_file, "r") as f:
+        supported_models = json.load(f)
+
+    if args.model not in supported_models:
+        raise ValueError(
+            f"Model {args.model} not supported. Please add it to {args.supported_models_file}."
+        )
+
+    model_path = supported_models[args.model]["model_path"]
+    api_base = supported_models[args.model]["api_base"]
+    token = supported_models[args.model]["token"]
+    context_length = supported_models[args.model]["context_length"]
```

Comment:
This single worker should be able to serve all models listed in `supported_models.json`.
It will register all workers to the controller and serve requests for all models.